library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationSource.java

```diff
@@ -89,17 +89,23 @@ void startSettingsApiResolutionForResult(Status status, Activity activity) throw
 
     @SuppressWarnings("ResourceType")
     void requestLocationUpdate() {
-        LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        if (isGoogleApiClientConnected()) {
```

Comment:
Thank you for your time to review. I am new in this kind of project kindly accept my apologies for the mistakes. I think same but without check connection status GoogleApiClient is throwing illegal state exception. With this check it is working fine.